src/core_ocean/shared/mpas_ocn_tracer_advection_mono.F

```diff
@@ -682,7 +682,7 @@ subroutine ocn_tracer_advection_mono_tend(tend, tracers, layerThickness,    &
             do k=3,kmax-1
               highOrderFlx(k, iCell) = (w(k,iCell)* &
                    (7.0_RKIND*(tracerCur(k  ,iCell) + tracerCur(k-1,iCell)) - &
-                              (tracerCur(k+1,iCell) + tracerCur(k-2,iCell))) - &
+                              (tracerCur(k+1,iCell) + tracerCur(k-2,iCell))) + &
```

Comment:
I agree this looks suspicious. I'll look more carefully tomorrow. One consideration is that k index is positive down and w is positive up. 

Does this equation appear in the sw or atm core for comparison? I didn't see it, with a quick check.

Reply:
@mark-petersen it is here in the atm core -- src/core_atmosphere/dynamics/mpas_atm_time_integration.F L3720 with the + instead of -

Reply:
Thanks for the reference to the atmospheric core. Those lines are here:
```
3717       flux4(q_im2, q_im1, q_i, q_ip1, ua) =                     &
3718           ua*( 7.*(q_i + q_im1) - (q_ip1 + q_im2) )/12.0
3719
3720       flux3(q_im2, q_im1, q_i, q_ip1, ua, coef3) =              &
3721                 flux4(q_im2, q_im1, q_i, q_ip1, ua) +           &
3722                 coef3*abs(ua)*((q_ip1 - q_im2)-3.*(q_i-q_im1))/12.0
```
Where the atmosphere `+` is at the end of 3721. I think the negative is correct for the ocean, because:
1. In the atmosphere, positive ua points with increasing k
2. In the ocean, positive w points with decreasing k

so `+abs(ua)` in the atmospheric stencil translates to `-abs(w)` in the ocean stencil. In other words, upwinding is in index space, but the direction of upwinding comes from the velocity. In the ocean the sign of the velocity is in physical space, which is opposite.

I can see the advantage of an ocean model that counts from the bottom up for stuff like this, but an upside-down k index seems to be a well-accepted standard in ocean models.

Here is a thought experiment: Make new variable `wDown=-w` for w pointed down, aligned with index space. If we do that everywhere, do we get the correct stencil relative to the S&G2011 paper?  Fluxes would also change sign, so `highOrderFlxDown = -highOrderFlx` etc is what we expect. I think the negative sign propagates through all the formulas, with the exception of `abs(w)` and `abs(wDown)`, which is why we see that negative sign:
```
abs(wDown) = - abs(w)
```
because in this case the absolute value clears the sign, and we have to put it back in manually.

Reply:
@mark-petersen I don't understand the logic here.  Both @ambrad and I worked from SG11 and defined derivatives appropriately relative to height, which seems correct.  It's not clear to me at all why it matters which way index space points.  If we define our derivatives appropriately that seems like the correct direction.  I have to think about this more.

Also note that looking at the copy pasted code, I think the atmospheric formulation is wrong.  The way it is written there should be a sign(ua) not abs(ua).  Ours is fine because we have ua*abs(ua) which is sign(ua).

Reply:
thinking about this further, I think you may be right.  I see how the index space direction matters here now.  But I'd like to write this out explicitly to fully convince myself.

Reply:
Also, note that `w = sign(w)*abs(w)` is the relation for comparing the paper to the code. The paper uses `sign(w)` and the code uses `abs(w)`. The lines are here:
```
 683               highOrderFlx(k, iCell) = (w(k,iCell)* &
 684                    (7.0_RKIND*(tracerCur(k  ,iCell) + tracerCur(k-1,iCell)) - &
 685                               (tracerCur(k+1,iCell) + tracerCur(k-2,iCell))) - &
 686                         coef3rdOrder*abs(w(k,iCell))* &
 687                              ((tracerCur(k+1,iCell) - tracerCur(k-2,iCell)) - &
 688                     3.0_RKIND*(tracerCur(k  ,iCell) - tracerCur(k-1,iCell))))/ &
 689                    12.0_RKIND
```
but I'll write it simpler with T as tracer and ignore iCell:
```
F(k) = w(k)*(      7*(T(k) + T(k-1)) - (T(k+1) + T(k-2))   ) &
       - coef3rdOrder*abs(w(k)) * ( T ...  etc)
```
The point here is that in SG2011, the first `u` is completely outside the square brackets, and multiplies all the way through:
![image](https://user-images.githubusercontent.com/4036816/97732698-885dc880-1a9c-11eb-9c7d-2cbcc611831f.png)
In the code, we multiply through by the u, and use `u*sign(u) = abs(u)` to get: 
```
F = u* (...) + abs(u)*(...)
```

Reply:
Mark, that is a good point concerning the sign w.r.t. positive w. In SG11, "sign(u) is positive for u indicating flow from cell i to cell i+1." In MPAS-O, I think you're saying positive w points up and k=1 is the surface, so the sign is already correct.